src/omega_moderne_client/client/moderne_client.py

```diff
@@ -140,6 +146,15 @@ async def run_campaign(self, campaign: Campaign, target_organization_id: str = "
         :param priority: The priority of the campaign. Can be one of "LOW" or "NORMAL".
         :return: The id of the recipe.
         """
+        # A Hacky Solution:
+        # The `runYamlRecipe` API endpoint can take longer than the Moderne gateway timeout of 60 seconds to respond.
+        # Particularly, when we are launching a recipe run against 30k+ repositories.
+        # When the gateway times out, we don't get the run ID back, so we need an alternate way to get it.
+        #
+        # As such, we embed our own custom 'uuid' in the recipe name, so that we can repeatedly poll the
+        # `previousRecipeRuns` API endpoint to search for our `uuid` and get the run ID,
+        # once the run has actually started.
```

Comment:
This comment describes the primary way I've attempted to fix this problem

Reply:
This is a typical solution for the problem: implementing a callback feature to determine the status of a request, while avoiding high network traffic or socket closures (time out). Another option would be to submit bulk sets of repos, instead of one large set.